### PR TITLE
Add hint about the return code of panic!

### PR DESCRIPTION
**Pull request by dns2utf8**
I hope the link works on all cases, since the `unreachable` doc is copied to `std::` as well.

### CODE REVIEW

**Comment by sfackler** [comment]
r? @steveklabnik 

cc @rust-lang/docs

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -24,6 +24,8 @@
 /// The multi-argument form of this macro panics with a string and has the
 /// `format!` syntax for building a string.
 ///
+/// If the main thread panics it will return with code `101`.
```
Comment (frewsxcv) [inline src/libstd/macros.rs]: Is this something that is guaranteed to happen across all platforms?

Comment (steveklabnik) [inline src/libstd/macros.rs]: it's currently hard-coded; I don't know if we want to commit to this

@rust-lang/lang

**Review by steveklabnik — COMMENTED** [review]
This seems fine but I'm not sure we want to lock into 101

**Comment by dns2utf8** [comment]
Is there a particular reason for this return code?
I could add a hint like "this is an implementaiton detail and may change"

**Comment by frewsxcv** [comment]
Something also to consider: although unusual, it's possible to catch panics, in which case this error code mention doesn't really apply

**Comment by withoutboats** [comment]
Seems like we can't change the default, people may be relying on it. Someday maybe we could let you set the error code, no idea how though.

**Comment by alexcrichton** [comment]
This seems like a case where we want to document the current behavior but not commit to it. We've done this a few other places in libstd, can we say what happens today and then explicitly state that it may change in the future?

**Comment by steveklabnik** [comment]
@alexcrichton yeah we've done

    /// # Current Implementation

in the past, could do that here as well. See https://doc.rust-lang.org/stable/std/primitive.slice.html#current-implementation

**Comment by steveklabnik** [comment]
@dns2utf8 ping! Any chance to update this PR according to the most recent comments?

**Comment by dns2utf8** [comment]
Yes, I will make it according to "current implementation"

**Comment by dns2utf8** [comment]
I think setting the error code could be done with the macro. Maybe something like this:
```
panic!("my {} panic", "fancy" -> 23);
```
Where 23 would be the returned error code. The `std::panic::set_hook` would then also require the function to return a number.

**Comment by dns2utf8** [comment]
I noticed something odd `std::sys::unix::abort_internal()` is implemented with `libc::abort()` which just exports the function from libc.

Using the following program but the return code is `134`. Do you know why?
```c
#include <stdlib.h>

int main() {

	// this will make the programm return 134 on linux-amd64 platform
	abort();

	return 42;
}

```

**Comment by joshtriplett** [comment]
@dns2utf8 134 is the return code for a SIGABRT:

```
/tmp$ (sleep 1000 ; echo return code: $?) &
[1] 3006
/tmp$ pidof sleep
3008
/tmp$ kill -ABRT 3008
bash: line 6:  3008 Aborted                 sleep 1000
return code: 134
[1]+  Done                    ( sleep 1000; echo return code: $? )
```

**Comment by dns2utf8** [comment]
Interesting, I did not know that.
Should I rebase the PR before merge? Would you like me to change something else?

**Comment by arielb1** [comment]
You have an overly-long line, please fix:

```
[00:02:23] tidy error: /checkout/src/libstd/macros.rs:29: line longer than 100 chars
```

I'll also ping @steveklabnik for a review.

**Pull request merged**